src/fmt.ml

```diff
@@ -187,6 +210,23 @@ let styled_string style = styled style string
 let stdout = Format.std_formatter
 let stderr = Format.err_formatter
 
+let with_strfmt f =
+  let buf = Buffer.create 17 in
+  let ppf = Format.formatter_of_buffer buf in
+  f ppf ;
+  Format.pp_print_flush ppf () ;
+  Buffer.contents buf
+
+let strfmt pp_v v = strf "%a" pp_v v
```

Comment:
OTOH why not ? `Fmt.{to,of}_to_string` seems pretty consitent with the OCaml conventions.